From 7b4a5eeb1e0e7521fb80246b0b162308157176d3 Mon Sep 17 00:00:00 2001

diff --git a/include/tlsh.h b/include/tlsh.h
index 5b8eeaa..c247e4e 100644
--- a/include/tlsh.h
+++ b/include/tlsh.h
@@ -60,11 +60,7 @@
 #ifndef HEADER_TLSH_H
 #define HEADER_TLSH_H
 
-#ifdef WINDOWS
-#include "win_version.h"
-#else
 #include "version.h"
-#endif
 
 #ifndef NULL
 #define NULL 0
diff --git a/include/tlsh_impl.h b/include/tlsh_impl.h
index 4309674..ea7922e 100644
--- a/include/tlsh_impl.h
+++ b/include/tlsh_impl.h
@@ -55,11 +55,7 @@
  * OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifdef WINDOWS
-#include "win_version.h"
-#else
 #include "version.h"
-#endif
 
 #ifndef HEADER_TLSH_IMPL_H
 #define HEADER_TLSH_IMPL_H
-- 
2.17.1

