From 38f365b03de3f2e2ca6a4d3428d4ce71363244e7 Mon Sep 17 00:00:00 2001
From: fab <fab@localhost>
Date: Thu, 4 Jul 2019 15:50:25 +0200
Subject: [PATCH] Remove dependency to WinFunction.h from tlsh.h


diff --git a/Windows/WinFunctions.h b/Windows/WinFunctions.h
index 31930d3..3873831 100644
--- a/Windows/WinFunctions.h
+++ b/Windows/WinFunctions.h
@@ -6,16 +6,6 @@
 #include <stdio.h>
 #include <time.h>
 
-#ifndef TLSH_LIB
-#  ifdef TLSH_EXPORTS
-#  define TLSH_API __declspec(dllexport)
-#  else
-#  define TLSH_API __declspec(dllimport)
-#  endif
-#else
-#  define TLSH_API
-#endif
-
 #define strdup _strdup
 #define NAME_LENGTH MAX_PATH
 #define snprintf _snprintf
diff --git a/include/tlsh.h b/include/tlsh.h
index 5b8eeaa..3efeace 100644
--- a/include/tlsh.h
+++ b/include/tlsh.h
@@ -112,7 +112,15 @@ class TlshImpl;
 #define TLSH_STRING_BUFFER_LEN (TLSH_STRING_LEN+1)
 
 #ifdef WINDOWS
-#include <WinFunctions.h>
+  #ifndef TLSH_LIB
+    #ifdef TLSH_EXPORTS
+      #define TLSH_API __declspec(dllexport)
+    #else
+      #define TLSH_API __declspec(dllimport)
+    #endif
+  #else
+    #define TLSH_API
+  #endif
 #else 
 	#if defined(__SPARC) || defined(_AS_MK_OS_RH73)
 	   #define TLSH_API
-- 
2.17.1

