diff --git a/src/parquet/schema.h b/src/parquet/schema.h
index 76920c0e..00105b9b 100644
--- a/src/parquet/schema.h
+++ b/src/parquet/schema.h
@@ -437,7 +437,10 @@ class PARQUET_EXPORT SchemaDescriptor {
   // -- -- b     |
   // -- -- -- c  |
   // -- -- -- -- d
-  std::unordered_map<int, const schema::NodePtr> leaf_to_base_;
+  //
+  // fix: error C2280: 'std::pair<_Kty,_Ty> &std::pair<_Kty,_Ty>::operator =...
+  // ...: attempting to reference a deleted function
+  std::unordered_map<int, schema::NodePtr> leaf_to_base_;

   // Mapping between ColumnPath DotString to the leaf index
   std::unordered_multimap<std::string, int> leaf_to_idx_;
