diff --git a/src/arrow/vendored/datetime/tz.cpp b/src/arrow/vendored/datetime/tz.cpp
index e05423e1..d533e355 100644
--- a/src/arrow/vendored/datetime/tz.cpp
+++ b/src/arrow/vendored/datetime/tz.cpp
@@ -202,7 +202,10 @@ get_known_folder(const GUID& folderid)
     if (SUCCEEDED(hr))
     {
         co_task_mem_ptr folder_ptr(pfolder);
-        folder = std::string(folder_ptr.get(), folder_ptr.get() + wcslen(folder_ptr.get()));
+        // fix: C4244: 'argument': conversion from '_Ty' to 'const _Elem', possible loss of data
+        // I guess they expect path to be ASCII
+        folder = std::string(reinterpret_cast<char*>(folder_ptr.get()),
+                             reinterpret_cast<char*>(folder_ptr.get() + wcslen(folder_ptr.get())));
     }
     return folder;
 }
