---
 cmake_modules/ThirdpartyToolchain.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake_modules/ThirdpartyToolchain.cmake b/cmake_modules/ThirdpartyToolchain.cmake
index 74f0e61c..70329e6d 100644
--- a/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cmake_modules/ThirdpartyToolchain.cmake
@@ -228,7 +228,7 @@ endif ()
 
 if (WIN32)
   ExternalProject_Add(tzdata_ep
-    URL "ftp://cygwin.osuosl.org/pub/cygwin/noarch/release/tzdata/tzdata-2019b-1.tar.xz"
+    URL "http://mirrors.kernel.org/sourceware/cygwin/noarch/release/tzdata/tzdata-2019b-1.tar.xz"
     URL_HASH MD5=17D8AB346BB1009D4EB35BC08040E51D
     CONFIGURE_COMMAND ""
     BUILD_COMMAND ""
-- 
2.17.1

