diff --git a/main.h b/main.h
index 621a531..e28aaf0 100644
--- a/main.h
+++ b/main.h
@@ -24,13 +24,33 @@
 # include "config.h"
 #endif
 
+#ifndef HAVE_FSEEKO
+#  ifdef _MSC_VER
+#    define _OFF_T_DEFINED
+#    include <stdint.h>
+typedef intmax_t _off_t;
+typedef _off_t off_t;
+#    ifdef _WIN64
+#      define ftello _ftelli64
+#      define fseeko _fseeki64
+#    else
+#      define ftello ftell
+#      define fseeko fseek
+#    endif  // ifdef _WIN64
+#   endif  // ifdef _MSC_VER
+#endif  // ifndef HAVE_FSEEKO
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
 #include <limits.h>
 #include <sys/stat.h>
-#include <unistd.h>
+
+#ifdef HAVE_UNISTD_H
+#  include <unistd.h>
+#endif
+
 #include <ctype.h>
 #include <inttypes.h>
 
-- 
2.17.1

