diff --git a/include/pybind11/detail/common.h b/include/pybind11/detail/common.h
index 892de0f..677300c 100644
--- a/include/pybind11/detail/common.h
+++ b/include/pybind11/detail/common.h
@@ -228,7 +228,7 @@ extern "C" {
     static PyObject *pybind11_init();                                          \
     PYBIND11_PLUGIN_IMPL(name) {                                               \
         int major, minor;                                                      \
-        if (sscanf(Py_GetVersion(), "%i.%i", &major, &minor) != 2) {           \
+        if (sscanf_s(Py_GetVersion(), "%i.%i", &major, &minor) != 2) {         \
             PyErr_SetString(PyExc_ImportError, "Can't parse Python version."); \
             return nullptr;                                                    \
         } else if (major != PY_MAJOR_VERSION || minor != PY_MINOR_VERSION) {   \
@@ -272,7 +272,7 @@ extern "C" {
     static void PYBIND11_CONCAT(pybind11_init_, name)(pybind11::module &);     \
     PYBIND11_PLUGIN_IMPL(name) {                                               \
         int major, minor;                                                      \
-        if (sscanf(Py_GetVersion(), "%i.%i", &major, &minor) != 2) {           \
+        if (sscanf_s(Py_GetVersion(), "%i.%i", &major, &minor) != 2) {         \
             PyErr_SetString(PyExc_ImportError, "Can't parse Python version."); \
             return nullptr;                                                    \
         } else if (major != PY_MAJOR_VERSION || minor != PY_MINOR_VERSION) {   \
