diff --git a/include/rapidjson/encodings.h b/include/rapidjson/encodings.h
index 50ad18bd..2b398ece 100644
--- a/include/rapidjson/encodings.h
+++ b/include/rapidjson/encodings.h
@@ -83,6 +83,10 @@ concept Encoding {
 \endcode
 */
 
+// Forward declaration.
+template<typename Stream>
+inline void PutUnsafe(Stream& stream, typename Stream::Ch c);
+
 ///////////////////////////////////////////////////////////////////////////////
 // UTF8
 
@@ -682,10 +686,6 @@ struct Transcoder {
     }
 };
 
-// Forward declaration.
-template<typename Stream>
-inline void PutUnsafe(Stream& stream, typename Stream::Ch c);
-
 //! Specialization of Transcoder with same source and target encoding.
 template<typename Encoding>
 struct Transcoder<Encoding, Encoding> {
